src/eventreceiver.ts

```diff
@@ -0,0 +1,47 @@
+import { Tracer } from './telemetry/traces';
+import { GlobalLogger as Logger } from './telemetry/logs';
+import { WorkflowFunction, WorkflowHandle, WorkflowParams } from './workflow';
+import { Transaction } from './transaction';
+import { MethodRegistrationBase } from './decorators';
+
+/*
+ * Info provided to an event receiver at initialization,
+ *  which contains the things that it needs to do its work
+ *  (retrieve decorated endpoints, and run new transactions / workflows)
+ */
+export interface DBOSExecutorEventReceiverInterface
+{
+  /* Logging service */
+  readonly logger: Logger;
+  /* Tracing service */
+  readonly tracer: Tracer;
+
+  getConfig<T>(key: string): T | undefined;
+  getConfig<T>(key: string, defaultValue: T): T;
+
+  /*
+   * Get the registrations for a receiver; this comes with:
+   *  minfo: the method info the receiver stored
+   *  cinfo: the class info the receiver stored
+   *  method: the method registration (w/ workflow, transaction, function, and other info)
+   */
+  getRegistrationsFor(eri: DBOSEventReceiver) : {minfo: unknown, cinfo: unknown, method: MethodRegistrationBase}[];
+
+  transaction<T extends unknown[], R>(txn: Transaction<T, R>, params: WorkflowParams, ...args: T): Promise<R>;
+  workflow<T extends unknown[], R>(wf: WorkflowFunction<T, R>, params: WorkflowParams, ...args: T): Promise<WorkflowHandle<R>>;
```

Comment:
Every time I do all 3 you guys ask me what is the use case... what is the use case for communicator?

Reply:
We didn't use communicators for Kafka because it is not safe (we commit the Kafka offset to indicate it's been processed). But I'm thinking this extensible system you build could have a bigger scope. For example, ideally someone can use this event receiving system to support web framework other than Koajs, or support other protocols like gRPC. Then we will need `external` or communicator.
Wdyt?

Reply:
So you think they're implementing a decorator like @GetApi?  Maybe I should be looking to see if that does anything else besides communicator.

Reply:
Oops sorry user getapi

Reply:
That would also involve `send`, `getEvent`, and `retrieveWorkflow`... thoughts?

Reply:
`send` and `getEvent`, etc are used by the actual handler/workflow context so probably not needed here.

Reply:
If you made a grpc server, would you not need either to construct instances of `HandlerContextImpl` or implement something similar to it?  In which case, you need more of `DBOSExecutor`, such as `send` and `getEvent`.  If we choose to make `HandlerContextImpl` a shared class, it needs to be implemented against the interface, which works with the exception of how it is handling `config` right now.

Reply:
I see your point here, but I think this PR could be scoped to support workflow/transaction/communicator first. Then we can think about how to extend this interface to allow users to implement other servers, which may require more thoughts.

Reply:
I put it in the interface already and don't see a reason to take it out.  Who knows what the end purpose will be...

I will not make `HandlerContextImpl` public, as this seems HTTP-specific.  If you wanted to do gRPC, you probably could, just write your own `HandlerContext` equivalent.